drake/systems/plants/@RigidBodyManipulator/doKinematics.m

```diff
@@ -99,6 +108,20 @@
   else
     kinsol.mex_ptr = options.kinematics_cache_ptr_to_use;
   end
+
+  if options.compute_gradients
+    nq = length(q);
+    nv = length(v);
+    if isnumeric(q)
+      q = TaylorVar(q, {[eye(nq), zeros(nq, nv)]});
```

Comment:
Yeah, that's the idea currently. We could perhaps use a struct with fields f and df to go back and forth between Matlab and C++ instead. As noted in the pull request description, there is also a bit of overhead in creating TaylorVars from mex as returned objects.